plsMD: A plasmid reconstruction tool for short-read assemblies
-----
Introduction
------------
Installation
-----------
```bash
conda install -c bioconda plsmd
```
------------
plsMD Single modality
----------
Preprocessing and Annotation
------
```bash
plsmd --preprocessing --dir <input_directory> --threads <num_threads> --db <plsdb_path>
```
| Option | Description | 
|---|---|
| `--dir` | Input directory containing unicylcer assembly FASTAs | 
| `--threads` | Number of threads | 
| `--db` | PLSDB database path | 
------
Processing and reconstruction
----
```bash
plsMD --process --dir <input_directory>
```
| Option | Description | 
|---|---|
| `--dir` | Input directory containing the preprocessed data (output from plsMD --preprocessing) | 
-------
plsMD batch modality
------
Phylogenetic Analysis
------
```bash
plsmd --phylogenetics --genes_dir <genes_directory> --min_length <min_length> --max_length <max_length> --threads <threads>
```
| Option | Description | 
|---|---|
| `--genes_dir` |Root directory containing gene folders. Each subdirectory should represent a gene family and contain the corresponding sequences | 
| `--threads` | Number of threads | 
| `--min_length` | Minimum sequence length for common sequence search. |
| `--min_length` | Maximum sequence length for common sequence search. |
---------
Output folders/files
| File/Directory | Description | Generated by |
|---|---|---|
| `<input_directory>/original_fasta/` | Contains the original FASTA files provided as input. | `plsMD --preprocessing` |
| `<input_directory>/<fasta_file>_abricate.tsv` | ABRICATE output files for each input FASTA file, detailing identified antimicrobial resistance genes. | `plsMD --preprocessing` |
| `<input_directory>/<fasta_file>_blastn.tsv` | BLASTn output files for each input FASTA file, showing matches against the PLSDB database. | `plsMD --preprocessing` |
| `<input_directory>/plasmid_list.txt` | A list of identified plasmid contigs. | `plsMD --preprocessing` |
| `<input_directory>/<fasta_file>_modified.fasta` | Modified FASTA files where plasmid contig names have been updated. | `plsMD --preprocessing` |
| `<input_directory>/rotated_fasta/` | Contains rotated FASTA files. | `plsMD --processing` |
| `<input_directory>/merged_fasta/` | Contains merged FASTA files. | `plsMD --processing` |
| `<input_directory>/plasmid_files/` | Contains processed plasmid contig files. | `plsMD --processing` |
| `<input_directory>/nonplasmid_files/` | Contains processed non-plasmid contig files. | `plsMD --processing` |
| `<input_directory>/nested_contigs.txt` |  (Add a description of this file) | `plsMD --processing` |
| `<input_directory>/nonreplicon_contigs/` | (Add a description of this directory and its contents) | `plsMD --processing` |
| `<input_directory>/Processed_ColContigs.txt` | Combined output file for processed Col and Inc plasmids. | `plsMD --processing` |
| `<genes_directory>/phylogenetic_tree/` | Contains the output of the phylogenetic analysis. | `plsMD --phylogenetics` |
| `<genes_directory>/phylogenetic_tree/<gene_name>_aligned.fasta` | Aligned FASTA files for each gene. | `plsMD --phylogenetics` |
| `<genes_directory>/phylogenetic_tree/<gene_name>_tree` | Phylogenetic trees in Newick format for each gene. | `plsMD --phylogenetics` |
| `<genes_directory>/gene_directories/` | Contains the gene directories used in the phylogenetic analysis | `plsMD --process` |
----------
